### PR TITLE
 Function test bugs fix: AgentCore dynamic installation and uninstallation

### PR DESCRIPTION
**Pull request by luanwenfei-venus**
【Fix issue】#1236
[Modification content] Repair 1. Repeated enhancement failure 2. Service is not closed during uninstallation 3. Logs are printed repeatedly 4. Error when uninstalling Agent 5. A single plug-in directory does not exist, causing all plug-in installations to fail 6. Not configuring dynamicPlugins will cause Agent startup exception 7 . After installing Sermant with different artifacts, only the configuration of the first installation will be read. 8. Chinese commas in the Agent startup log. 9. After uninstalling and reinstalling, the core service will not take effect.
[Use case description] Not involved
[Self-test situation] 1. Local static inspection and cleaning situation 2. Test case execution passed
[Scope of influence] Not involved

### CODE REVIEW

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/common/LoggerFactory.java [hunk 1]
```diff
@@ -59,6 +69,14 @@ public static void init() {
      * @return jul日志
      */
     public static Logger getLogger() {
-        return logger;
+        if (sermantLogger != null) {
+            return sermantLogger;
+        }
+
+        // 日志如果重复获取将会有同样的日志重复打
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/common/LoggerFactory.java]: 注释描述的有点不清楚

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginSystemEntrance.java [hunk 2]
```diff
@@ -66,11 +68,16 @@ public static void initialize(boolean isDynamic) {
...
             // 初始化支持动态安装的主动启动插件 agentmain方式启动时执行
-            Set<String> activePlugins = pluginSetting.getDynamicPlugins().get("active");
-            if (activePlugins == null || activePlugins.isEmpty()) {
+            Map<String, Set<String>> dynamicPlugins = pluginSetting.getDynamicPlugins();
+            if (MapUtils.isEmpty(dynamicPlugins)) {
                 LOGGER.info("Non dynamic-support-plugin is configured to be loaded.");
                 return;
             }
+            Set<String> activePlugins = dynamicPlugins.get("active");
+            if (activePlugins == null || activePlugins.isEmpty()) {
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginSystemEntrance.java]: 使用工具类

File: sermant-agentcore/sermant-agentcore-premain/src/main/java/com/huaweicloud/sermant/premain/common/AgentArgsResolver.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        if (agentArgs == null) {
+            return argsMap;
+        }
+        for (String arg : agentArgs.trim().split(",")) {
+            final int index = arg.indexOf('=');
+            if (index >= 0) {
+                argsMap.put(arg.substring(0, index).trim(), arg.substring(index + 1).trim());
+            }
+        }
+        return argsMap;
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-premain/src/main/java/com/huaweicloud/sermant/premain/common/AgentArgsResolver.java]: 判断条件应该是index>0, index=0的话key是空的

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/com/huaweicloud/sermant/implement/service/tracing/sender/TracingSender.java [hunk 4]
```diff
@@ -88,7 +87,11 @@ public void start() {
         }
         this.isSending = true;
         gatewayClient = ServiceManager.getService(GatewayClient.class);
-        EXECUTOR.execute(new SpanEventSendThread());
+        if (executorService == null || executorService.isShutdown()) {
+            executorService = Executors.newSingleThreadExecutor(
+                    runnable -> new Thread(runnable, "tracing-sender-thread"));
+        }
+        executorService.execute(new SpanEventSendThread());
```
Comment (hanbingleixue) [inline sermant-agentcore/sermant-agentcore-implement/src/main/java/com/huaweicloud/sermant/implement/service/tracing/sender/TracingSender.java:94]: executorService.execute(new SpanEventSendThread());应该只有executorService == null || executorService.isShutdown()的时候执行吧

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginSystemEntrance.java (hunk 2 above)
Comment (luanwenfei-venus) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginSystemEntrance.java]: 已修改

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/common/LoggerFactory.java (hunk 1 above)
Comment (luanwenfei-venus) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/common/LoggerFactory.java]: 已修改

**Review by lilai23 — DISMISSED** [review]

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/com/huaweicloud/sermant/implement/service/tracing/sender/TracingSender.java (hunk 4 above)
Comment (luanwenfei-venus) [reply sermant-agentcore/sermant-agentcore-implement/src/main/java/com/huaweicloud/sermant/implement/service/tracing/sender/TracingSender.java:94]: 这里不会，上面判断了这个服务的flag

**Review by TangLeDaily — DISMISSED** [review]

**Review by TangLeDaily — APPROVED** [review]

**Review by daizhenyu — APPROVED** [review]

**Pull request merged**